src/common/meta/src/ddl/create_flow_task.rs

```diff
@@ -0,0 +1,242 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod check;
+mod metadata;
+
+use std::collections::BTreeMap;
+
+use api::v1::flow::flow_request::Body as PbFlowRequest;
+use api::v1::flow::{CreateRequest, FlowRequest};
+use async_trait::async_trait;
+use common_procedure::error::{FromJsonSnafu, ToJsonSnafu};
+use common_procedure::{
+    Context as ProcedureContext, LockKey, Procedure, Result as ProcedureResult, Status,
+};
+use common_telemetry::info;
+use futures::future::join_all;
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+use strum::AsRefStr;
+use table::metadata::TableId;
+
+use super::utils::add_peer_context_if_needed;
+use crate::ddl::utils::handle_retry_error;
+use crate::ddl::DdlContext;
+use crate::error::Result;
+use crate::key::flow_task::flow_task_info::FlowTaskInfoValue;
+use crate::key::FlowTaskId;
+use crate::lock_key::{CatalogLock, FlowTaskNameLock};
+use crate::peer::Peer;
+use crate::rpc::ddl::CreateFlowTask;
+use crate::{metrics, ClusterId};
+
+/// The procedure of flow task creation.
+pub struct CreateFlowTaskProcedure {
```

Comment:
Perhaps `CreateFlowProcedure` is already clear enough.

```suggestion
pub struct CreateFlowProcedure {
```

Reply:
> Perhaps `CreateFlowProcedure` is already clear enough.

The remaining renaming(e.g., `FlowTaskMetadataManager`) will be tracked in #3664